kolibri/core/public/api.py

```diff
@@ -146,3 +156,90 @@ def get_public_file_checksums(request, version):
         json.dumps({"id": error_constants.NOT_FOUND, "metadata": {"view": ""}}),
         content_type="application/json",
     )
+
+
+def position_in_queue(id):
+    qs = SyncQueue.objects.all().order_by("datetime").values("id")
```

Comment:
Hopefully the queue is never that big so querying the whole queue wouldn't be an issue, but could possibly do this with a query to get the datetime for the id, and then just query the count of items that have a datetime less than that?

Reply:
I don't think performance will be a problem here, if the queue gets thousands of request we'll have to look for different problems . 
In any case, there's an `updated` field https://github.com/jredrejo/kolibri/blob/sync_endpoint/kolibri/core/device/models.py#L198 that's a float number and should be as quick as the id when sorting. Adding an index to `updated` and ordering by it should be more or less as performant as what you propose with less code (and likely less db queries) . Does it sounds good ?

Reply:
Done